src/Interpreters/IInterpreterUnionOrSelectQuery.h

```diff
@@ -12,8 +12,13 @@ class IInterpreterUnionOrSelectQuery : public IInterpreter
 {
 public:
     IInterpreterUnionOrSelectQuery(const ASTPtr & query_ptr_, ContextPtr context_, const SelectQueryOptions & options_)
+        : IInterpreterUnionOrSelectQuery(query_ptr_, Context::createCopy(context_), options_)
+    {
+    }
+
+    IInterpreterUnionOrSelectQuery(const ASTPtr & query_ptr_, ContextMutablePtr context_, const SelectQueryOptions & options_)
         : query_ptr(query_ptr_)
-        , context(Context::createCopy(context_))
+        , context(context_)
```

Comment:
Sometimes we already have mutable context (e.g. `InterpreterFactory::get`). There is no need to create a copy.

Reply:
>  it looks to me like the original context is supposed to stay unchanged.

It should be fine. There are lots of interpreters take mutable reference of context, e.g. InterpreterDropQuery.